### PR TITLE
check for source lexeme with no length moved to the beginning of the class factory

### PR DESCRIPTION
**Pull request by zproksi**

### CODE REVIEW

File: srcbpatch/streamreplacer.cpp [hunk 1]
```diff
@@ -517,35 +509,30 @@ unique_ptr<StreamReplacer> EqualLengthReplacer(StreamReplacerChoice& choice, con
...
 }
 
 
 //--------------------------------------------------
 unique_ptr<StreamReplacer> StreamReplacer::CreateReplacer(StreamReplacerChoice& choice)
 {
+    for (AbstractLexemesPair& alpair: choice)
+    {
+        if (alpair.first->access().size() < 1)
+            throw logic_error("Pattern to replace cannot be empty");
```
Comment (VladimirTakeda) [inline srcbpatch/streamreplacer.cpp:533]: try
 if (alpair.first->access().empty())

File: srcbpatch/streamreplacer.cpp [hunk 2]
```diff
@@ -517,35 +509,30 @@ unique_ptr<StreamReplacer> EqualLengthReplacer(StreamReplacerChoice& choice, con
...
     }
 
-    if (bSameSize)
-    {
-        return EqualLengthReplacer(choice, szSrc); // create optimized replacer for lexemes of the same length
-    }
-
-    return unique_ptr<StreamReplacer>(new ChoiceReplacer(choice));
+    return EqualLengthReplacer(choice, szSrc); // create optimized replacer for lexemes of the same length
 }
```
Comment (VladimirTakeda) [inline srcbpatch/streamreplacer.cpp:524]: EqualLengthReplacer function has a name like a class. I was confused and checked what the EqualLengthReplacer is.

Comment (VladimirTakeda) [reply srcbpatch/streamreplacer.cpp:524]: I am used to adding a verb at the beginning of function. And the std library do the same
std::MAKE_unique
std::FIND_if
std:: GETline

File: srcbpatch/streamreplacer.cpp [hunk 3]
```diff
@@ -517,35 +509,30 @@ unique_ptr<StreamReplacer> EqualLengthReplacer(StreamReplacerChoice& choice, con
...
 
 
 //--------------------------------------------------
 unique_ptr<StreamReplacer> StreamReplacer::CreateReplacer(StreamReplacerChoice& choice)
 {
```
Comment (VladimirTakeda) [inline srcbpatch/streamreplacer.cpp:529]: It's unclear for me, why StreamReplacer has the CreateReplacer method. I wound create a factory or just a factory namespace (namespace ReplacerFactory).

Comment (zproksi) [reply srcbpatch/streamreplacer.cpp:529]: It is for sacred wars beginning.
There are people who think that class factory must be in the separate namespace, and those who think that class can hold the way to be created inside itself.
You can chose dark/light side and/or can follow your own approach.

Wound be addressed here.

File: srcbpatch/streamreplacer.cpp (hunk 2 above)
Comment (zproksi) [reply srcbpatch/streamreplacer.cpp:524]: Addressed.

File: srcbpatch/streamreplacer.cpp (hunk 1 above)
Comment (zproksi) [reply srcbpatch/streamreplacer.cpp:533]: fair note. Addressed.

**Pull request merged**